### PR TITLE
Fix minTS and maxTS reading NAN bug

### LINKED ISSUE #24: Bug: minTS and maxTS can read as NAN
More details described in [SigViewer #124](https://github.com/cbrnr/sigviewer/issues/124)

### PR DESCRIPTION
**Pull request by Yida-Lin**
Fix #24  NAN reading bug. More details in [SigViewer #124](https://github.com/cbrnr/sigviewer/issues/124)

### CODE REVIEW

File: xdf.cpp [hunk 1]
```diff
@@ -735,22 +735,22 @@ void Xdf::findMinMax()
     //find the smallest timestamp of all streams
     for (auto const &stream : streams)
     {
-        if (stream.info.first_timestamp != -1)
+        if (!std::isnan(stream.info.first_timestamp) && stream.info.first_timestamp != -1)
```
Comment (Yida-Lin) [inline xdf.cpp]: To be honest, I have no idea why I put `-1` here back then. Seems NAN check is needed here, but the -1 can be taken away? Advice appreciated.

File: xdf.cpp [hunk 2]
```diff
@@ -735,22 +735,22 @@ void Xdf::findMinMax()
...
         {
             minTS = stream.info.first_timestamp;
             break;
         }
     }
     for (auto const &stream : streams)
     {
-        if (stream.info.first_timestamp != -1 && stream.info.first_timestamp < minTS)
+        if (!std::isnan(stream.info.first_timestamp) && stream.info.first_timestamp != -1 && stream.info.first_timestamp < minTS)
```
Comment (Yida-Lin) [inline xdf.cpp]: Same here.

File: xdf.cpp [hunk 3]
```diff
@@ -735,22 +735,22 @@ void Xdf::findMinMax()
...
             minTS = stream.info.first_timestamp;
     }
 
     //find the max timestamp of all streams
     for (auto const &stream : streams)
     {
-        if (stream.info.last_timestamp != 0 && stream.info.last_timestamp > maxTS)
+        if (!std::isnan(stream.info.last_timestamp) && stream.info.last_timestamp != 0 && stream.info.last_timestamp > maxTS)
```
Comment (Yida-Lin) [inline xdf.cpp]: Same here, not sure why checking `stream.info.last_timestamp != 0`.

File: xdf.cpp (hunk 1 above)
Comment (cbrnr) [reply xdf.cpp]: No idea why the `-1` and `0` values. There was a discussion in #8 regarding this topic, but I can't remember the details.

Comment (Yida-Lin) [reply xdf.cpp]: @cbrnr Yeah looks like a mistake made back then. I have removed these. Tested with a few files and seems working fine.

Comment (tstenner) [reply xdf.cpp]: No idea either, but the lsl timestamps are guaranteed to be positive.

**Comment by Yida-Lin** [comment]
I am merging this one as this fixes a known bug and we have consensus here. Please let me know if otherwise.

**Pull request merged**